ansible_navigator/documentation_generation/doc_updater.py

```diff
@@ -0,0 +1,279 @@
+""" documentation updater
+"""
+import difflib
+import logging
+import os
+import sys
+import tempfile
+
+from argparse import ArgumentParser
+from argparse import ArgumentDefaultsHelpFormatter
+from argparse import Namespace
+
+from copy import copy
+from shutil import copyfile
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Tuple
+from typing import Union
+
+import yaml
+
+
+from ..configuration_subsystem import NavigatorConfiguration
+from ..configuration_subsystem.definitions import Entry
+
+from ..utils import oxfordcomma
+
+logger = logging.getLogger()
+
+APP = "ansible-navigator"
+
+PARAM_HEADER = (
+    "Name",
+    "Description",
+    "Default",
+    "Choices",
+    "CLI paramters",
+    "Environment variable",
+    "Settings file (ansible-navigator.)",
+)
+RST_TABLE_HEADER = [
+    ".. list-table:: {}",
+    "  :header-rows: 1",
+]
+RST_FIRST_ROW_ENTRY = "  * - {}"
+RST_ADDITONAL_ROW_ENTRY = "    - {}"
+
+
+def _file_diff(current: str, should_be: str):
+    """diff 2 files"""
+    with open(current, "r") as current_f:
+        with open(should_be, "r") as should_f:
+            diff = difflib.unified_diff(
+                current_f.read().splitlines(),
+                should_f.read().splitlines(),
+                fromfile="current",
+                tofile="should_be",
+            )
+    return list(diff)
+
+
+def _rst_generate_row(row: Tuple) -> List:
+    """generate an r for an rst list table"""
+    data = []
+    data.append(RST_FIRST_ROW_ENTRY.format(row[0]))
+    for row_part in row[1:]:
+        data.append(RST_ADDITONAL_ROW_ENTRY.format(row_part))
```

Comment:
If the value is `None` for example id `choices` is not part of subcommand it can be omitted in the final display